google/cloud/spanner/samples/samples.cc

```diff
@@ -527,6 +527,126 @@ void FieldAccessOnNestedStruct(google::cloud::spanner::Client client) {
+  }
+  //! [serialize-query-partition]
+}
+
+std::string Receive() { return std::string(); }  // NOLINT
+
+//! [deserialize-read-partition]
+google::cloud::StatusOr<google::cloud::spanner::ReadPartition>
+ReceiveReadPartitionFromRemoteMachine() {
+  std::string serialized_partition = Receive();
+  return google::cloud::spanner::DeserializeReadPartition(serialized_partition);
+}
+//! [deserialize-read-partition]
+
+//! [deserialize-query-partition]
+google::cloud::StatusOr<google::cloud::spanner::QueryPartition>
+ReceiveQueryPartitionFromRemoteMachine() {
+  std::string serialized_partition = Receive();
+  return google::cloud::spanner::DeserializeQueryPartition(
+      serialized_partition);
+}
+//! [deserialize-query-partition]
+
+void ProcessRow(google::cloud::spanner::Row<std::int64_t, std::string,
+                                            std::string>) {  // NOLINT
+}
+
+void PartitionRead(google::cloud::spanner::Client client) {
+  namespace spanner = google::cloud::spanner;
+  //! [key-set-builder]
+  auto key_set = spanner::KeySetBuilder<spanner::Row<int64_t>>()
+                     .Add(spanner::MakeKeyRangeClosed(spanner::MakeRow(1),
+                                                      spanner::MakeRow(10)))
+                     .Build();
+  //! [key-set-builder]
+
+  //! [partition-read]
+  spanner::Transaction ro_transaction = spanner::MakeReadOnlyTransaction();
+  google::cloud::StatusOr<std::vector<spanner::ReadPartition>> partitions =
+      client.PartitionRead(ro_transaction, "Singers", key_set,
+                           {"SingerId", "FirstName", "LastName"});
+  if (partitions) {
+    for (auto& partition : *partitions) {
+      SendPartitionToRemoteMachine(partition);
+    }
+  }
+  //! [partition-read]
+
+  //! [read-read-partition]
+  google::cloud::StatusOr<spanner::ReadPartition> partition =
+      ReceiveReadPartitionFromRemoteMachine();
+  if (partition) {
+    auto result_set = client.Read(*partition);
+    if (result_set) {
+      for (auto& row :
+           result_set->Rows<std::int64_t, std::string, std::string>()) {
+        if (row) {
+          ProcessRow(*row);
+        }
+      }
+    }
+  }
+  //! [read-read-partition]
+}
+
+void PartitionQuery(google::cloud::spanner::Client client) {
+  namespace spanner = google::cloud::spanner;
+  //! [partition-query]
+  spanner::Transaction ro_transaction = spanner::MakeReadOnlyTransaction();
+  google::cloud::StatusOr<std::vector<spanner::QueryPartition>> partitions =
+      client.PartitionQuery(
+          ro_transaction,
+          spanner::SqlStatement(
+              "select SingerId, FirstName, LastName from Singers"));
+  if (partitions.ok()) {
+    for (auto& partition : *partitions) {
+      SendPartitionToRemoteMachine(partition);
+    }
+  }
+  //! [partition-query]
+
+  //! [execute-sql-query-partition]
+  google::cloud::StatusOr<spanner::QueryPartition> partition =
+      ReceiveQueryPartitionFromRemoteMachine();
+  if (partition) {
```

Comment:
I think this is easier to grok if you do:

```C++
if (!partition) throw std::runtime_error(parition.status().message());
```

That reduces nesting and is similar to the error handling in other examples.